demo/src/screens/componentScreens/SegmentedControlScreen.tsx

```diff
@@ -17,32 +17,56 @@ const segments = {
   forth: [{label: 'With'}, {label: 'Custom'}, {label: 'Style'}],
   fifth: [{label: 'Full'}, {label: 'Width'}],
   sixth: [{label: 'Full'}, {label: 'Width'}, {label: 'With'}, {label: 'A'}, {label: 'Very Long Segment'}],
-  seventh: [{label: '$'}, {label: '%'}]
+  seventh: [{label: '$'}, {label: '%'}],
+  eighth: [
+    {label: `${Assets.emojis.arrow_left} Left`},
+    {label: `${Assets.emojis.arrow_up} Up`},
+    {label: `${Assets.emojis.arrow_down} Down`},
+    {label: `${Assets.emojis.arrow_right} Right`}
+  ]
 };
 
 const SegmentedControlScreen = () => {
   const onChangeIndex = useCallback((index: number) => {
     console.warn('Index ' + index + ' of the second segmentedControl was pressed');
   }, []);
 
+  const [screenPreset, setScreenPreset] = useState(SegmentedControl.presets.DEFAULT);
+
   return (
     <View flex bottom padding-page>
       <Text center text40 $textDefault>
         Segmented Control
       </Text>
-      <View flex marginT-s8>
+      <View
+        flex
+        marginT-s8
+        key={screenPreset /* added here because changing the preset based on state was causing wrong calculations */}
```

Comment:
Why it was cousing wrong calculations?

Reply:
I'm not sure why but when the preset changes based on state the outlines sometimes gets the dimensions and offsets worng.

Reply:
So we should check why. Maybe some props that are not updated and should. This is not a good solution since the probable scenario is using it with a state change

Reply:
Inbal, I discussed this with Nitzan
the component initialize all sizes and keep it as shared values so the easiest way to solve it is by passing a key but because it's not a real use case (there's no reason to change the preset in runtime) I don't think we should invest time on it
We have other examples screen where we use the "key" trick, so it's ok IMO

Reply:
The real use case will be when the editor will offer the user to select one of the two presets. I think we should try to fix this issue. May be reset the values if the preset changes or at leat do the key hack in the component and not ask the user to do it... just throwing ideas, I didn't check it myself yet

Reply:
@ethanshar Do you think we should add the key thing inside the component?

Reply:
Very nice fix!